astropy/coordinates/sky_coordinate.py

```diff
@@ -1659,11 +1659,11 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
 
         Returns
         -------
-        vcorr : `~astropy.units.Quantity` with velocity units
+        vcorr : `~astropy.units.Quantity`
             The  correction with a positive sign.  I.e., *add* this
             to an observed radial velocity to get the barycentric (or
             heliocentric) velocity. If m/s precision or better is needed,
-            see the notes below.
+            see the notes below. Must have physical type of 'speed'.
```

Comment:
Continuing my out of scope comment from above (😅 ), if we decided to go down the path of extending numpydoc to support expected physical types, this would become:
```python
vcorr : `~astropy.units.Quantity` [speed]
```
But ignore this for now!!

Reply:
I really like this. One of my other PRs (#10662, to which I need to return) is very compatible with this notation.
The python ``typing``module has eg `list[str]` as the constructor of the static type "list of str", so as #10662 and #11200 move us in that direction, I think many parameter annotations can be expressed in the typing-style: e.g., ``vcorr : `~astropy.units.Quantity`[speed]`` (note the missing space), and ``flag: str or array_like[str]``

Reply:
Agreed that extending numpydoc would help this reach its maxima utility, but we could achieve a very similar effect by adding all the physical types to ``docs/conf.py::numpydoc_xref_aliases``, pointing each to their physical type class (when #11200 is merged). Now glossary additions required!

Reply:
I see that specific implementation might not work, but something similar, surely, like including the list of units nd corresponding physical types in the docs, and x-ref to that.